File: src/main/java/org/activecraft/activecraftcore/guis/profilemenu/item/TimeItem.java
package org.activecraft.activecraftcore.guis.profilemenu.item;

import org.activecraft.activecraftcore.guicreator.GuiItem;
import org.activecraft.activecraftcore.guicreator.GuiNavigator;
import org.activecraft.activecraftcore.guis.profilemenu.inventory.ReasonsProfile;
import org.activecraft.activecraftcore.messagesv2.MessageFormatter;
import org.activecraft.activecraftcore.messagesv2.MessageSupplier;
import kotlin.Pair;
import org.activecraft.activecraftcore.guicreator.GuiItem;
import org.activecraft.activecraftcore.guicreator.GuiNavigator;
import org.activecraft.activecraftcore.guis.profilemenu.inventory.ReasonsProfile;
import org.activecraft.activecraftcore.messagesv2.MessageFormatter;
import org.activecraft.activecraftcore.messagesv2.MessageSupplier;
import org.bukkit.Material;

public class TimeItem extends GuiItem {

    public TimeItem(int time, String timeString, ReasonsProfile reasonsProfile) {
        super(Material.CLOCK);
        MessageSupplier messageSupplier = reasonsProfile.getMessageSupplier();
        setDisplayName(messageSupplier.getColorScheme().primary() + timeString);
        setLore(messageSupplier.getFormatted("profile.reasons-gui.set-time",
                new MessageFormatter(messageSupplier.getActiveCraftMessage(), new Pair<>("time", time + ""))));
        addClickListener(guiClickEvent -> {
            //ReasonsProfilev2 reasonsProfilev2 = (ReasonsProfile) guiClickEvent.getGui().getGuiCreator();
            reasonsProfile.setBanTime(time);
            reasonsProfile.select(37, 44, guiClickEvent.getSlot());
            GuiNavigator.pushReplacement(reasonsProfile.getPlayer(), reasonsProfile.build());
        });
    }
}

File: src/main/java/org/activecraft/activecraftcore/guis/profilemenu/item/TimeItem.kt
package org.activecraft.activecraftcore.guis.profilemenu.item

import org.activecraft.activecraftcore.guicreator.GuiClickEvent
import org.activecraft.activecraftcore.guicreator.GuiItem
import org.activecraft.activecraftcore.guicreator.GuiNavigator.Companion.pushReplacement
import org.activecraft.activecraftcore.guis.profilemenu.inventory.ReasonsProfile
import org.activecraft.activecraftcore.messages.MessageFormatter
import org.activecraft.activecraftcore.messages.MessageSupplier
import org.bukkit.Material

class TimeItem(time: Int, timeString: String, reasonsProfile: ReasonsProfile) : GuiItem(Material.CLOCK) {
    init {
        val messageSupplier: MessageSupplier = reasonsProfile.messageSupplier
        setDisplayName(messageSupplier.colorScheme.primary.toString() + timeString)
        setLore(
            messageSupplier.getFormatted(
                "profile.reasons-gui.set-time",
                MessageFormatter(messageSupplier.activeCraftMessage, Pair("time", time.toString() + ""))
            )
        )
        addClickListener { guiClickEvent: GuiClickEvent ->
            reasonsProfile.banTime = time
            reasonsProfile.select(37, 44, guiClickEvent.slot)
            pushReplacement(reasonsProfile.player, reasonsProfile.build())
        }
    }
}